### PR TITLE
Use PassManager in addInitGuards, localizeGlobals, bulkCopyRecords, and removeUnnecessaryAutoCopyCalls

### PR DESCRIPTION
**Pull request by aconsroe-hpe**
Continues #19164

This begins a refactor on passes in addInitGuards, localizeGlobals, bulkCopyRecords, and removeUnnecessaryAutoCopyCalls. See commits for more commentary on a per-pass basis, but generally my approach was:

1) isolate the use of the global ast vecs
2) convert file statics into class statics
3) convert forv_* as appropriate (see #19240 for why this is complicated)
4) leave comments about other global state that could be an issue

As ever, this is an active WIP so I feel okay leaving so many comments because I anticipate answering a lot of the questions as the PassManager effort progresses.

Testing
* [x] paratest
* [x] paratest with --fast
* [x] paratest with comm

### CODE REVIEW

File: compiler/optimizations/localizeGlobals.cpp [hunk 1]
```diff
@@ -39,56 +42,65 @@
...
-    for_vector(BaseAST, ast, asts) {
-      if (SymExpr* se = toSymExpr(ast)) {
-        Symbol* var = se->symbol();
-        ModuleSymbol* parentmod = toModuleSymbol(var->defPoint->parentSymbol);
-        CallExpr* parentExpr = toCallExpr(se->parentExpr);
-        bool inAddrOf = parentExpr && (parentExpr->isPrimitive(PRIM_ADDR_OF) || parentExpr->isPrimitive(PRIM_SET_REFERENCE));
-        bool lhsOfMove = parentExpr && isMoveOrAssign(parentExpr) && (parentExpr->get(1) == se);
+void LocalizeGlobals::process(FnSymbol* fn) {
+  // REFACTOR Q -- keeping these around is nice for re-use, but maybe we should
+  // have a standard method like ::cleanup/teardown
```
Comment (dlongnecke-cray) [inline compiler/optimizations/localizeGlobals.cpp:47]: 👍

File: compiler/optimizations/removeUnnecessaryAutoCopyCalls.cpp [hunk 2]
```diff
@@ -27,83 +27,84 @@
...
 // and corresponding calls to chpl__autoDestroy for Plain-Old-Data (POD) types.
 // See isPOD() and propagateNotPOD().
-static void removePODinitDestroy()
+
+//
+// remove pointless initCopy calls, e.g., initCopy calls on records of
+// primitive types
+//
+
+bool RemoveUnnecessaryAutoCopyCalls::shouldProcess(FnSymbol* fn) {
```
Comment (dlongnecke-cray) [inline compiler/optimizations/removeUnnecessaryAutoCopyCalls.cpp:39]: Wow, this is much easier to read!

File: compiler/passes/normalize.cpp [hunk 3]
```diff
@@ -4588,6 +4588,7 @@ static void find_printModuleInit_stuff() {
     if (symbol->hasFlag(FLAG_PRINT_MODULE_INIT_INDENT_LEVEL)) {
       gModuleInitIndentLevel = toVarSymbol(symbol);
       INT_ASSERT(gModuleInitIndentLevel);
+      // Q why  doesn't this break out of the loop
```
Comment (dlongnecke-cray) [inline compiler/passes/normalize.cpp:4591]: 🤔

**Review by dlongnecke-cray — APPROVED** [review]
I think this PR really demonstrates that this approach is the right direction. The new passes are immediately easier to read (even though "not much has changed" other than the reorg).

File: compiler/passes/normalize.cpp (hunk 3 above)
Comment (dlongnecke-cray) [reply compiler/passes/normalize.cpp:4591]: I don't know if you want to try adding a `break` and then running tests, but I can't imagine we have more than one symbol with that flag.

Comment (aconsroe-hpe) [reply compiler/passes/normalize.cpp:4591]: Would you prefer I try to tackle and answer that now? I left it rhetorical with the intention of addressing it when attacking normalize.cpp, but could investigate further now too

Comment (dlongnecke-cray) [reply compiler/passes/normalize.cpp:4591]: Oh are you planning on catching that variable during normalize or something instead? I'm fine with either way, I'm not confident a `break` will make much of a difference (though I do think it's appropriate).

Comment (aconsroe-hpe) [reply compiler/passes/normalize.cpp]: I pushed a commit that answers this question. It didn't turn out to be complicated so I'm including it in this PR

Comment (dlongnecke-cray) [reply compiler/passes/normalize.cpp]: Ah, ok, this makes sense. Thanks for taking the time to answer this!

Comment (aconsroe-hpe) [reply compiler/passes/normalize.cpp]: Thanks for the push to answer it!

**Pull request merged**